tests/unit/Rules/CountryCodeTest.php

```diff
@@ -30,59 +36,33 @@ public function testShouldThrowsExceptionWhenInvalidFormat(): void
         new CountryCode('whatever');
     }
 
-    public function testShouldUseISO3166Alpha2ByDefault(): void
-    {
-        $country = new CountryCode();
-        self::assertEquals(CountryCode::ALPHA2, $country->set);
-    }
-
-    public function testShouldDefineACountryFormatOnConstructor(): void
-    {
-        $country = new CountryCode(CountryCode::NUMERIC);
-        self::assertEquals(CountryCode::NUMERIC, $country->set);
-    }
-
-    public function providerForValidCountryCode()
-    {
-        return [
-            [CountryCode::ALPHA2,  'BR'],
-            [CountryCode::ALPHA3,  'BRA'],
-            [CountryCode::NUMERIC, '076'],
-            [CountryCode::ALPHA2,  'DE'],
-            [CountryCode::ALPHA3,  'DEU'],
-            [CountryCode::NUMERIC, '276'],
-            [CountryCode::ALPHA2,  'US'],
-            [CountryCode::ALPHA3,  'USA'],
-            [CountryCode::NUMERIC, '840'],
-        ];
-    }
-
-    public function providerForInvalidCountryCode()
-    {
-        return [
-            [CountryCode::ALPHA2,  'USA'],
-            [CountryCode::ALPHA3,  'US'],
-            [CountryCode::NUMERIC, '000'],
-        ];
-    }
-
     /**
-     * @dataProvider providerForValidCountryCode
+     * {@inheritdoc}
      */
-    public function testValidContryCodes($format, $input): void
+    public function providerForValidInput(): array
     {
-        $rule = new CountryCode($format);
-
-        self::assertTrue($rule->validate($input));
+        return [
+            [new CountryCode(CountryCode::ALPHA2),  'BR'],
+            [new CountryCode(CountryCode::ALPHA3),  'BRA'],
+            [new CountryCode(CountryCode::NUMERIC), '076'],
+            [new CountryCode(CountryCode::ALPHA2),  'DE'],
+            [new CountryCode(CountryCode::ALPHA3),  'DEU'],
+            [new CountryCode(CountryCode::NUMERIC), '276'],
+            [new CountryCode(CountryCode::ALPHA2),  'US'],
+            [new CountryCode(CountryCode::ALPHA3),  'USA'],
+            [new CountryCode(CountryCode::NUMERIC), '840'],
```

Comment:
I don't know, this `CountryCode(CountryCode::FOO)` getting the param from the same class seems not good 😕

Reply:
you mean the constants `NUMERIC`, `ALPHA2`, and `ALPHA3`? I think it's fine.